tests/user.js

```diff
@@ -0,0 +1,193 @@
+          username: 'user',
+          first: 'first',
+          last: 'last',
+          email: 'user@gmail.com',
+          password: 'user',
+          role: 'viewer'
+        })
+        .expect({message: 'No such role exists'}, done);
+    });
+  });
+
+  describe('/users/:user_id CRUD operations', function () {
+    it('finds a user with matching id', function (done) {
+      request
+        .get('/api/users/' + userId)
+        .set('x-access-token', token)
+        .expect(200)
+        .end(function (err, res) {
+          expect(res.body).to.exist;
+          expect(res.body).to.be.an('object');
+          expect(res.body).to.have.keys('_id', 'email', 'username', 'name', 'role', '__v');
+          done();
+        });
+    });
+
+    it('find a non-existent user', function (done) {
+      request
+        .get('/api/users/57d11f35b0a303c1234569df')
+        .set('x-access-token', token)
+        .expect(404)
+        .expect({message: 'User not found'}, done);
+    });
+
+    it('updates a particular user data', function (done) {
+      request
+        .put('/api/users/' + userId)
+        .set('x-access-token', token)
+        .send({
+          username: 'njerry-werry',
+          first: 'njerry',
+          last: 'werry',
+          password: 'njeri',
+          email: 'njeri@gmail.com'
+        })
+        .end(function (err, res) {
+          expect(res.status).to.equal(200);
+          expect(res.body.message).to.equal('User has been updated');
+          done();
+        });
+    });
+
+    it('updating a user that doesnt exist', function (done) {
+      request
+        .put('/api/users/57d11f35b0a303c1234569df')
+        .set('x-access-token', token)
+        .send({
+          username: 'njerry',
+        })
+        .end(function (err, res) {
+          expect(res.status).to.equal(404);
+          expect(res.body.message).to.equal('User not found');
+          done();
+        });
+    });
+
+    it('returns all documents belonging to a particular user', function (done) {
+      request
+        .get('/api/users/' + userId + '/documents')
+        .set('x-access-token', token)
+        .end(function (err, res) {
+          expect(res.body).to.exist;
+          expect(Array.isArray(res.body)).to.equal(true);
+          expect(res.body[0]).to.have.keys('_id', 'title', 'ownerId', 'role', 'accessLevel', 'content', 'modifiedAt', 'createdAt', '__v');
+          done();
+        });
+    });
+
+    it('asserts that cannot find documents o a non-existent user', function (done) {
+      request
+        .get('/api/users/57d11f35b0a303c1234569df/documents')
+        .set('x-access-token', token)
+        .expect(404)
+        .expect({message: 'User not found'}, done);
+    });
+
+    it('deletes a particular user inforation', function (done) {
+      request
+        .delete('/api/users/' + userId)
+        .set('x-access-token', token)
+        .expect({message: 'User has been deleted'}, done);
+    });
+  });
+});
```

Comment:
You should add a test to confirm that someone cannot edit another person's account as implemented in the code.